### PR TITLE
Atombot fixes, msgpack documentation

### PR DESCRIPTION
**Pull request by plipay**
Some fixes to Atombot, including:
- Explicit use of serialization, which was otherwise failing. I've also made this clearer with the comments and added another stream to show writing binary data.
- Using locks to protect against race conditions on updates to the position/representation vars

Also updated the documentation regarding msgpack, pointing out it defaults to True, not False, and explaining how to turn it off/on.

https://github.com/elementary-robotics/atom/issues/152

### CODE REVIEW

File: doc/source/includes/walkthrough.md [hunk 1]
```diff
@@ -362,6 +404,11 @@ Now that our atombot element is running, let's interact with it using `atom-cli`
     - `command atombot move_left 2`
 - We can control the rate at which the messages are printed by passing in the rate value (hz)
     - `read atombot pos_map 1`
+- By default, atom-cli automatically msgpacks the data that is sent/received
```
Comment (dpipemazo) [inline doc/source/includes/walkthrough.md]: :+1: nice! Great comment

File: utilities/atom-cli/atom-cli.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def __init__(self):
 
             "cmd_msgpack": cleandoc("""
                 Turns on/off msgpack serialization and deserialization.
-                Pass True to use msgpack, False to turn off. Default False.
+                Pass True to use msgpack, False to turn off. Defaults to True.
```
Comment (dpipemazo) [inline utilities/atom-cli/atom-cli.py]: :+1: nice catch

File: languages/python/tests/test_atom.py [hunk 3]
```diff
@@ -213,7 +213,7 @@ def test_read_since(self, caller, responder):
         entries = caller.entry_read_since("test_responder", "test_stream", last_id, 2)
         assert(len(entries) == 2)
         assert entries[-1]["data"] == b"1"
-        
+
```
Comment (dpipemazo) [inline languages/python/tests/test_atom.py]: :+1: nice catch. Not sure if your style formatter got this or if you caught it manually.

**Comment by plipay** [comment]
Agree about the multiple files, it bothers me too. I've knocked it down from 3 to 2 by making the one inside examples a symlink. However, I don't see an easy way to embed the source code file inside of the markdown tutorial. A quick search online mostly yields people begging for this functionality and getting ignored by the markdown devs (: . For now will stick with 2 files

**Pull request merged**